plugins/ssr/ssrcommon.ml

```diff
@@ -1440,7 +1440,20 @@ end
 
 let return ~orig_name:_ ~new_name:_ = tclUNIT ()
 
-let tclINTRO_ID id = tclINTRO ~id:(Some id) ~conclusion:return
+
+
+let tclINTRO_ID_ltac ist = function
```

Comment:
This ugly code can be removed by using GADTs, as I did here: https://github.com/LPCIC/elpi/blob/master/src/elpi_API.mli#L507 (actually it would be simpler here). The FFI then looks like:  https://github.com/LPCIC/elpi/blob/master/src/elpi_builtin.ml#L476 where the ML function has a type described by the GADT. Do we have anything like that in Coq already? @ppedrot ?